deepchem/feat/base_classes.py

```diff
@@ -159,59 +147,60 @@ class MolecularFeaturizer(Featurizer):
   uses SMILES strings and RDKIT molecule objects to represent
   small molecules. All other featurizers which are subclasses of
   this class should plan to process input which comes as smiles
-  strings or RDKIT molecules. 
+  strings or RDKIT molecules.
 
   Child classes need to implement the _featurize method for
   calculating features for a single molecule.
 
-  Note
-  ----
-  In general, subclasses of this class will require RDKit to be installed.
+  Notes
+  -----
+  The subclasses of this class require RDKit to be installed.
   """
 
-  def featurize(self, molecules, log_every_n=1000):
+  def featurize(self, molecules, log_every_n=1000, canonical=False):
     """Calculate features for molecules.
 
     Parameters
     ----------
-    molecules: RDKit Mol / SMILES string / iterable
-        RDKit Mol, or SMILES string or iterable sequence of RDKit mols/SMILES
-        strings.
+    molecules : RDKit Mol / SMILES string / iterable
+      RDKit Mol, or SMILES string or iterable sequence of RDKit mols/SMILES
+      strings.
+    log_every_n : int, default 1000
+      Logging messages reported every `log_every_n` samples.
+    canonical : bool, default False
+      Whether to use a canonical order of atoms returned by RDKit
 
     Returns
     -------
-    A numpy array containing a featurized representation of
-    `datapoints`.
+    features : np.ndarray
+      A numpy array containing a featurized representation of `datapoints`.
     """
     try:
       from rdkit import Chem
-      from rdkit.Chem import rdmolfiles
-      from rdkit.Chem import rdmolops
       from rdkit.Chem.rdchem import Mol
     except ModuleNotFoundError:
       raise ValueError("This class requires RDKit to be installed.")
+
     # Special case handling of single molecule
     if isinstance(molecules, str) or isinstance(molecules, Mol):
       molecules = [molecules]
     else:
       # Convert iterables to list
       molecules = list(molecules)
+
     features = []
     for i, mol in enumerate(molecules):
       if i % log_every_n == 0:
         logger.info("Featurizing datapoint %i" % i)
       try:
-        # Process only case of SMILES strings.
         if isinstance(mol, str):
           # mol must be a SMILES string so parse
           mol = Chem.MolFromSmiles(mol)
-          # TODO (ytz) this is a bandage solution to reorder the atoms
-          # so that they're always in the same canonical order.
-          # Presumably this should be correctly implemented in the
-          # future for graph mols.
-          if mol:
-            new_order = rdmolfiles.CanonicalRankAtoms(mol)
-            mol = rdmolops.RenumberAtoms(mol, new_order)
+        # canonicalize
+        if canonical:
+          canonical_smiles = Chem.MolToSmiles(mol)
+          mol = Chem.MolFromSmiles(canonical_smiles)
```

Comment:
I add canonicalize option to `MolecularFeaturizer`

Reply:
Yes. Please confirm this documents.
https://www.rdkit.org/docs/GettingStartedInPython.html#writing-molecules

I think the official canonicalize approach is like this.

```
>>> Chem.MolToSmiles(Chem.MolFromSmiles('C1=CC=CN=C1'))
'c1ccncc1'
>>> Chem.MolToSmiles(Chem.MolFromSmiles('c1cccnc1'))
'c1ccncc1'
>>> Chem.MolToSmiles(Chem.MolFromSmiles('n1ccccc1'))
'c1ccncc1'
```

Reply:
Ok looks good! Feel free to merge in once ready

Reply:
I noticed this implementation loses coordinate information if mol object has them ...
So, I went back to original implementation.